test/unitTests/core/zoe/contracts/test-simpleOfferAuction.js

```diff
@@ -0,0 +1,252 @@
+import { test } from 'tape-promise/tape';
+import harden from '@agoric/harden';
+
+import { makeZoe } from '../../../../../core/zoe/zoe/zoe';
+import { makeSimpleOfferMaker } from '../../../../../core/zoe/contracts/simpleOffer/simpleOffer';
+import { makeSecondPriceSrcs } from '../../../../../core/zoe/contracts/simpleOffer/srcs/secondPriceSrcs';
+import { setup } from '../setupBasicMints';
+
+test('zoe.makeInstance with simpleOffer with secondPriceSrcs', async t => {
+  try {
+    const { issuers: originalIssuers, mints, assays } = setup();
+    const issuers = originalIssuers.slice(0, 2);
+    const zoe = makeZoe();
+    const escrowReceiptIssuer = zoe.getEscrowReceiptIssuer();
+
+    // Setup Alice
+    const aliceMoolaPurse = mints[0].mint(issuers[0].makeAmount(1));
+    const aliceMoolaPayment = aliceMoolaPurse.withdrawAll();
+    const aliceSimoleanPurse = mints[1].mint(issuers[1].makeAmount(0));
+    const aliceSimoleanPayment = aliceSimoleanPurse.withdrawAll();
+
+    // Setup Bob
+    const bobMoolaPurse = mints[0].mint(issuers[0].makeAmount(0));
+    const bobMoolaPayment = bobMoolaPurse.withdrawAll();
+    const bobSimoleanPurse = mints[1].mint(issuers[1].makeAmount(11));
+    const bobSimoleanPayment = bobSimoleanPurse.withdrawAll();
+
+    // Setup Carol
+    const carolMoolaPurse = mints[0].mint(issuers[0].makeAmount(0));
+    const carolMoolaPayment = carolMoolaPurse.withdrawAll();
+    const carolSimoleanPurse = mints[1].mint(issuers[1].makeAmount(7));
+    const carolSimoleanPayment = carolSimoleanPurse.withdrawAll();
+
+    // Setup Dave
+    const daveMoolaPurse = mints[0].mint(issuers[0].makeAmount(0));
+    const daveMoolaPayment = daveMoolaPurse.withdrawAll();
+    const daveSimoleanPurse = mints[1].mint(issuers[1].makeAmount(5));
+    const daveSimoleanPayment = daveSimoleanPurse.withdrawAll();
+
+    // 1: Alice creates a secondPriceAuction instance
+    const secondPriceSrcs = makeSecondPriceSrcs(3);
+    const makeSecondPriceAuction = makeSimpleOfferMaker(secondPriceSrcs);
```

Comment:
I don't understand why the packaging is done this way. Alice puts the simpleOfferMaker and secondPrice code together.  Would it make sense to have an auctionMaker that did that packaging, so Alice doesn't have to?

Reply:
Yeah, I agree that it's rough. I think the "packaging" issue is related to how Alice can reuse highly-vetted governing contracts from libraries. That is, if Alice just looks up a name, this will turn out pretty different. Is it ok if we punt this until we can have a discussion on how to share reusable code, use it easily, and convince other people that you used a particular library?

Reply:
certainly.